src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -539,6 +569,32 @@ export class StudyViewPageStore
             )
         );
 
+        this.reactionDisposers.push(
```

Comment:
this could arguably be done in onresult handler of the molecularProfileNameSet promise.